frontend/public/components/app.jsx

```diff
@@ -311,41 +311,63 @@ const PollConsoleUpdates = React.memo(function PollConsoleUpdates() {
 
   const [isToastOpen, setToastOpen] = React.useState(false);
   const [pluginsChanged, setPluginsChanged] = React.useState(false);
+  const [pluginVersionsChanged, setPluginVersionsChanged] = React.useState(false);
   const [consoleChanged, setConsoleChanged] = React.useState(false);
   const [isFetchingPluginEndpoints, setIsFetchingPluginEndpoints] = React.useState(false);
   const [allPluginEndpointsReady, setAllPluginEndpointsReady] = React.useState(false);
 
   const [pluginsData, setPluginsData] = React.useState();
   const [pluginsError, setPluginsError] = React.useState();
+  const [pluginManifestsData, setPluginManifestsData] = React.useState();
+  const [pluginManifestsError, setPluginManifestsError] = React.useState();
   const safeFetch = React.useCallback(useSafeFetch(), []);
-  const tick = React.useCallback(() => {
+  const updatesTick = React.useCallback(() => {
     safeFetch(`${window.SERVER_FLAGS.basePath}api/check-updates`)
       .then((response) => {
         setPluginsData(response);
         setPluginsError(null);
       })
       .catch(setPluginsError);
   }, [safeFetch]);
-  usePoll(tick, URL_POLL_DEFAULT_DELAY);
+  usePoll(updatesTick, URL_POLL_DEFAULT_DELAY);
+  const fetchPluginManifest = (pluginName) =>
+    coFetchJSON(`${window.SERVER_FLAGS.basePath}api/plugins/${pluginName}/plugin-manifest.json`);
+  const manifestsTick = React.useCallback(() => {
+    const pluginManifests = pluginsData?.plugins?.map((pluginName) =>
+      fetchPluginManifest(pluginName),
+    );
+    Promise.all(pluginManifests)
+      .then((response) => {
+        setPluginManifestsData(response);
+        setPluginManifestsError(null);
+      })
+      .catch(setPluginManifestsError);
```

Comment:
If any of the plugin services are down, we'll stop checking updates for any of them... It would be better to handle errors individually so that we can still use the version data we have for other plugins.